### ISSUE TITLE
One-Sided Procrustes for Toeplitz matrices

### ISSUE THREAD

**Issue by PaulWAyers** [theochem/procrustes#84]
The Procrustes method can be solved where the transformation is a Toeplitz matrix
    http://dx.doi.org/10.1155/2013/696019
I don't know any applications of this, so I consider it low priority. But it does look straightforward.

**Comment by PaulWAyers** [comment]
This can also be used for Hankel matrices. 

A key question is whether the "closest Toeplitz matrix" or "closest Hankel matrix" to a given matrix is useful? Or are there cases wehre the transformation as a Hankel/Toeplitz matrix.

**Comment by PaulWAyers** [comment]
This issue is "easy" (it seems) but I don't see an application, so I've marked it low priority.

**Comment by PK-SS** [comment]
I want to debug this error. Please guide me for furthur things.

**Comment by FanwangM** [comment]
Thank you for your interest. I would assume you want to learn how to start. You can go through our Procrustes paper (https://doi.org/10.1016/j.cpc.2022.108334) as well as this one (http://dx.doi.org/10.1155/2013/696019). You can get familiar with the general background and the related math details for one-sided Procrustes for Toeplitz matrices problem. Then you can start to get familiar with the API design and implement your codes. But please be sure to fork our repo and work on a new branch instead of the master/main branch. Once you think the code is ready for review, please make a pull request and tag me. I will take a look.

Thank you and feel free to post details questions in GitHub Actions.
@PK-SS

**Comment by PaulWAyers** [comment]
@PK-SS are you still interested in this?

**Comment by Ao-chuba** [comment]
Hello @PaulWAyers @FanwangM,
I have been looking into the codebase for a while now and from my understanding, the goal is to implement Procrustes solutions where the transformation matrix is constrained to be a Toeplitz or Hankel matrix, following the approach described in [Here](http://dx.doi.org/10.1155/2013/696019). 
what I plan todo is :

> Create a new module `procrustes/toeplitz.py` to keep the logic modular (similar to `orthogonal.py`).
> Implement `toeplitz() `and `hankel()` solver functions based on the paper's algorithm.
> Add unit tests in `test_toeplitz.py `to verify correctness against known cases.

I would like to work on this issue if required. Please let me know if this approach looks correct

**Comment by FanwangM** [comment]
Hi @Ao-chuba . Thanks for your insterest in contributing. Your plan is sensible. Please feel free to make a PR. But to make it clear, we don't require anyone to work on the issues, but we welcome community contributions. Cheers.
